streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -474,6 +492,7 @@ public void shouldThrowTaskMigratedExceptionOnSubsequentCallWhenProducerFencedIn
                 " indicating the task may be migrated out; it means all tasks belonging to this thread should be migrated.")
         );
 
+        collector.send(topic, "3", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
```

Comment:
These tests used to rely on the fact that the `sendException` was never forgotten by just setting it once and then asserting that multiple subsequent calls also threw it. So now we need to call `send` before each to re-insert the exception

Reply:
For this particular test, considering the name, it seem the test is void now?

Reply:
Maybe I misinterpreted this, but I took the `OnSubsequentCall` in the name to mean that it would throw on the next (ie subsequent) call after the _send_, not that it would continue to throw on all subsequent calls. ie I think it should actually be several different tests (one for each "call" that should throw) but got mashed into just one

Reply:
I guess I should just break these up into different tests then, huh. Will do

Reply:
Not 100% sure how `OnSubsequentCall` is meant either. But what you say seems to make sense and thus it should be different test. Thanks for the extra mile splitting them up!